Libraries/MLXVLM/MediaProcessing.swift

```diff
@@ -0,0 +1,157 @@
+// Copyright © 2024 Apple Inc.
+
+import CoreImage.CIFilterBuiltins
+import MLX
+import MLXLMCommon
+
+private let context = CIContext()
+
+/// Collection of methods for processing media (images, video, etc.).
+///
+/// A typical image preparation pipeline might look like this:
+///
+/// ```swift
+/// var image: CIImage
+/// image = MediaProcessing.inSRGBToneCurveSpace(image)
+///
+/// // apply user instructions
+/// image = MediaProcessing.apply(image, processing: processing)
+///
+/// image = MediaProcessing.resampleBicubic(image, to: config.size.cgSize)
+/// image = MediaProcessing.normalize(
+///     image, mean: config.imageMeanTuple, std: config.imageStdTuple)
+///
+/// return MediaProcessing.asMLXArray(image)
+/// ```
```

Comment:
The paligemma media processing pipeline.  On the python side this is spread across mlx-vlm, transformers and in some cases some tensor flow libraries.